dmutils/html.py

```diff
@@ -0,0 +1,21 @@
+from jinja2 import Markup, escape
+
+
+def link_to(url, label='', **attrs):
+    """Create a hyperlink with the given text pointing to the URL"""
+
+    if url == '' or not isinstance(url, str):
+        raise ValueError('link_to expects URL to be a non-empty string')
+
+    if label == '' or label is None:
+        label = url
+
+    output = ['<a']
+    output.append(f' href="{escape(url)}"')
+    for key, value in attrs.items():
+        if isinstance(value, str):
+            output.append(' {}="{}"'.format(escape(key), escape(value)))
+    output.append('>')
+    output.append(escape(label))
+    output.append('</a>')
+    return Markup("".join(output))
```

Comment:
I like what you've done here, building the string up in a list and then using `join()` is a good idea as it's usually quicker and more flexible than appending to a string (in Python at least).

That said, I think this would be more readable if you relied more heavily on the power of Python f-strings. What I would do is assemble the attrs as a separate string using a list, and then create the entire element with one f-string. Something like

```
<a href={escape(url)}{attrs}>{escape(label)}</a>
```